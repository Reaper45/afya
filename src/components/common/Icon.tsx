import * as React from 'react'

import styled from 'lib/emotion'

const IconWrapper = styled("span")<{ active?: boolean }>`
  color: ${props =>
    props.active ? props.theme.colors.accent : props.theme.colors.primary};
`;

const Icon: React.FC<{
  active?: boolean; icon: JSX.Element;
}> = ({ active, icon}) => {
  return <IconWrapper active={active}>{icon}</IconWrapper>;
};

export default Icon;

export const calenderIcon = (
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="24" height="24">
    <path className="heroicon-ui" d="M17 4h2a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V6c0-1.1.9-2 2-2h2V3a1 1 0 1 1 2 0v1h6V3a1 1 0 0 1 2 0v1zm-2 2H9v1a1 1 0 1 1-2 0V6H5v4h14V6h-2v1a1 1 0 0 1-2 0V6zm4 6H5v8h14v-8z"/>
  </svg>
)

export const chatIcon = (
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="24" height="24">
    <path className="heroicon-ui" d="M6 14H4a2 2 0 0 1-2-2V4c0-1.1.9-2 2-2h12a2 2 0 0 1 2 2v2h2a2 2 0 0 1 2 2v13a1 1 0 0 1-1.7.7L16.58 18H8a2 2 0 0 1-2-2v-2zm0-2V8c0-1.1.9-2 2-2h8V4H4v8h2zm14-4H8v8h9a1 1 0 0 1 .7.3l2.3 2.29V8z"/>
  </svg>
)

export const moreHorizontalIcon = (
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="24" height="24">
    <path className="heroicon-ui" d="M4 15a3 3 0 1 1 0-6 3 3 0 0 1 0 6zm0-2a1 1 0 1 0 0-2 1 1 0 0 0 0 2zm8 2a3 3 0 1 1 0-6 3 3 0 0 1 0 6zm0-2a1 1 0 1 0 0-2 1 1 0 0 0 0 2zm8 2a3 3 0 1 1 0-6 3 3 0 0 1 0 6zm0-2a1 1 0 1 0 0-2 1 1 0 0 0 0 2z"/>
  </svg>
)