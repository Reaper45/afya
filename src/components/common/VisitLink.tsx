import React from 'react'
import { NavLink } from 'react-router-dom'

import styled from 'lib/emotion'
import Icon from "./Icon";

import menu from "static/icons/menu.svg";

const VisitLinkWrapper = styled(NavLink)`
  padding: .5rem 1rem;
  background: ${props => props.theme.colors.light};
  display: flex;
  flex-direction: row;

`

const VisitLink: React.FC<{ count: number; title: string }> = ({
  count,
  title
}) => {
  return (
    <VisitLinkWrapper to="">
      <span data-name="count">{count}</span>
      <span data-name="title">{title}</span>
      <Icon src={menu} />
    </VisitLinkWrapper>
  );
};

export default VisitLink
