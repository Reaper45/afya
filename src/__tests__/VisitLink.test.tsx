import React from "react";
import { shallow } from "enzyme";

import VisitLink from "components/common/VisitLink";
import Icon from "components/common/Icon";

import icon from "static/icons/menu.svg";

const visit: { count: number; title: string } = { count: 4, title: "Kiambu" };

it("renders without crashing", () => {
  const component = shallow(<VisitLink {...visit} />);
  expect(component).toMatchSnapshot();
});

it("renders props correctly", () => {
  const component = shallow(<VisitLink {...visit} />);
  expect(component.find('[data-name="count"]').text()).toEqual("4");
  expect(component.find('[data-name="title"]').text()).toEqual("Kiambu");
});

it("contains more menu", () => {
  const component = shallow(<VisitLink {...visit} />);
  expect(component.containsMatchingElement(<Icon src={icon} />)).toBe(true);
});
