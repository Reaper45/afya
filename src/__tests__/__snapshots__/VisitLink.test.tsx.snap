// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Styled(NavLink)
  to=""
>
  <span
    data-name="count"
  >
    4
  </span>
  <span
    data-name="title"
  >
    Kiambu
  </span>
  <Icon
    src="menu.svg"
  />
</Styled(NavLink)>
`;
